backend/locationService.ts

```diff
@@ -0,0 +1,44 @@
+import { getLongLatData } from "./positionApiService";
+import type {
+  LocalBusStopInformation,
+  Position,
+  StationInformation,
+} from "../src/types";
+import { getBusStopsWithinRadius } from "./tflApiService";
+import { handleLatestArrivalsRequest } from "./busArrivalsService";
+
+export async function handlePostcodeRequest(
+  postcode: string
+): Promise<StationInformation[]> {
+  const longLatData: Position = await getLongLatData(postcode);
+  const position: Position = {
+    latitude: longLatData.latitude,
+    longitude: longLatData.longitude,
+  };
+
+  const allLocalBusStops: LocalBusStopInformation[] =
+    await getBusStopsWithinRadius(position.latitude, position.longitude);
+  const orderedLocalBusStops: LocalBusStopInformation[] =
+    orderBusStopData(allLocalBusStops);
+  const closestLocalBusStops: LocalBusStopInformation[] =
+    limitClosesBusStops(orderedLocalBusStops);
+
+  const networkArrivals: StationInformation[] = await Promise.all(
+    closestLocalBusStops.map(async (busStop) => {
+      const arrivals = await handleLatestArrivalsRequest(busStop.naptanId);
+      return arrivals;
+    })
+  );
+  return networkArrivals;
+}
+
+function orderBusStopData(localBusStops: LocalBusStopInformation[]) {
+  return [...localBusStops].sort((a, b) => (a.distance < b.distance ? -1 : 1));

```

Comment:
Could use
`return [...localBusStops].sort((a, b) => a.distance - b.distance);`
See examples in [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)